submissions/factorio_draftsman/main.md

```diff
@@ -0,0 +1,211 @@
+
+### Factorio Movie Player Resurrection
+
+A little while ago, while trawling for projects to use as examples for this very article, I came across the classic [Factorio Sandstorm](https://www.youtube.com/watch?v=mgfwwqwxdxY). A perfect project to adapt; or, it would have been, if I could actually load the map in-game! The original map version was still dated to the tender version `0.14.20`. In addition, a lot of the migrations that took place over such a long span of time broke the functionality of the script to convert the image frames into map data, as well as some functionality of the map itself, so just downloading an old version of Factorio and coercing it along wasn't going to be enough to revive the old save. For example, the `build.lua` script in charge of encoding still called the `automation-science-pack` and `logistic-science-pack` items `science-pack-1` and `science-pack-2`. That should give you some indication of how old the map was!
+
+It troubled me, seeing such an iconic piece of Factorio history fall into disrepair. So took the time to fix the issues and migrate it all the way to 1.1.57:
+
+| ![video of feel the beat](./img/cut_and_shrunk.gif) |
+| ------------------------ |
+| Pictured is part of [Feel the Beat](https://youtu.be/aLZQ-0dHbiU?t=38), keeping the tradition of Darude music videos. |
+
+Changing the build script could be done by hand, but the signal `raw-wood` that the machine internally used no longer exists in modern Factorio. In order to fix it, I replaced all occurrences of the signal with it in the map with `artillery-wagon` (since I knew it would be unique), and that map updating was done with a script running Draftsman code. I also added a number of other scripts for extracting the images from the source, as well as taking screenshots of the result and stitching them into an output video, which pretty easy in Python. I used said scripts to make the output above.
+
+I was also playing around with the idea of using Draftsman to make a blueprint-based version of the builder script instead of the console-based original method; this would allow you to see exactly what parts of the memory you are modifying by placing a grid aligned blueprint on top of them, and would also allow you to extend the memory of the machine by simply adding more blocks (the memory provided on the map is only enough for 4800 frames). I figured this was less important than updating what was there originally, especially when the code that *was* already there I knew worked. There could be other considerations as well; the amount of encoded data is pretty hefty, especially for blueprint strings which I suspect may be less dense than the console scripts, which are so large they are dumped in text files when built. Determining whether a blueprint method was actually feasible I figured could wait until I had some spare time further down the line.
```

Comment:
```suggestion
I was also playing around with the idea of using Draftsman to make a blueprint-based version of the builder script, instead of the console-based original method; this would allow you to see exactly what parts of the memory you are modifying by placing a grid aligned blueprint on top of them, and would also allow you to extend the memory of the machine by simply adding more blocks (the memory provided on the map is only enough for 4800 frames). I figured this was less important than updating what was there originally, especially when the code that *was* already there I knew worked. There could be other considerations as well; the amount of encoded data is pretty hefty, especially for blueprint strings which I suspect may be less dense than the console scripts, which are so large they are dumped in text files when built. Determining whether a blueprint method was actually feasible I figured could wait until I had some spare time further down the line.
```